ydb/library/yql/providers/generic/provider/yql_generic_list.cpp

```diff
@@ -0,0 +1,281 @@
+    *request.mutable_selects()->Add() = select;
+
+    auto promise = NThreading::NewPromise();
+    handles.emplace_back(promise.GetFuture());
+
+    Y_ENSURE(State_->GenericClient);
+
+    State_->GenericClient->ListSplits(request).Subscribe([desc, promise, tableAddress]
+        (const NConnector::TListSplitsStreamIteratorAsyncResult f3) mutable {
+        NConnector::TListSplitsStreamIteratorAsyncResult f4(f3);
+        auto streamIterResult = f4.ExtractValueSync();
+
+        // Check transport error
+        if (!streamIterResult.Status.Ok()) {
+            desc->Issues.AddIssue(TStringBuilder()
+                << "Call ListSplits for table " << tableAddress.ToString() << ": "
+                << streamIterResult.Status.ToDebugString());
+            promise.SetValue();
+            return;
+        }
+
+        Y_ENSURE(streamIterResult.Iterator);
+
+        auto drainer =
+            NConnector::MakeListSplitsStreamIteratorDrainer(std::move(streamIterResult.Iterator));
+
+        // pass drainer to the callback because we want him to
+        // stay alive until the callback is called    
+        drainer->Run().Subscribe([desc, promise, tableAddress, drainer]
+            (const NThreading::TFuture<NConnector::TListSplitsStreamIteratorDrainer::TBuffer>& f5) mutable {
+            NThreading::TFuture<NConnector::TListSplitsStreamIteratorDrainer::TBuffer> f6(f5);
+            auto drainerResult = f6.ExtractValueSync();
+
+            // check transport and logical errors
+            if (drainerResult.Issues) {
+                TIssue dstIssue(TStringBuilder() << "Call ListSplits for table " << tableAddress.ToString());
+
+                for (const auto& srcIssue : drainerResult.Issues) {
+                    dstIssue.AddSubIssue(MakeIntrusive<TIssue>(srcIssue));
+                };
+
+                desc->Issues.AddIssue(std::move(dstIssue));
+                promise.SetValue();
+                return;
+            }
+
+            // collect all the splits from every response into a single vector
+            for (auto&& response : drainerResult.Responses) {
+                std::transform(std::make_move_iterator(response.mutable_splits()->begin()),
+                               std::make_move_iterator(response.mutable_splits()->end()),
+                               std::back_inserter(desc->Splits),
+                               [](auto&& split) { return std::move(split); });
+            }
+
+            promise.SetValue();
+        });
+    });
+
+    return TIssues();
+}
+
+IGraphTransformer::TStatus TGenericListTransformer::DoApplyAsyncChanges(TExprNode::TPtr input, TExprNode::TPtr& output, TExprContext& ctx) {
+    AsyncFuture_.GetValue();
+    output = input;
+
+    const auto& reads = FindNodes(input, [&](const TExprNode::TPtr& node) {
+        if (const auto maybeRead = TMaybeNode<TGenReadTable>(node)) {
+            return maybeRead.Cast().DataSource().Category().Value() == GenericProviderName;
```

Comment:
У `TGenReadTable` на сколько я знаю обязательно должен быть провайдер `GenericProviderName`, можно просто вернуть `true` (или дополнительно проверить):

```suggestion
            Y_ENSURE(maybeRead.Cast().DataSource().Category().Value() == GenericProviderName);
            return true;
```